### ISSUE TITLE
Support TATE prods (-> enable display rotation)

### ISSUE THREAD

**Issue by oh-ren** [openMSX/openMSX#1054]
Would be great (though not a big issue) if openMSX could support TATE games/demos (e.g. [Destroyer](https://www.msx.org/news/software/en/msxdev14-8-destroyer) & Impact's _Impaccen en Wegwezen_ (any other examples?)) by allowing to rotate the screen.
In both examples the screen has to be rotated 90° CCW, though I figure CW should be supported as well.

Command proposal:
`set display_rotate <degrees-or-identifier>`

While degrees could technically be anything, only `0`, `-90`  &  `90` make sense. These values could be aliased by `off`, `left` & `right` respectively

Though (a wider point of discussion) I do find having to use `set <setting> <value>` all the time a bit verbose/cumbersome, so an alias could be:
`tate (left|right|off)` and/or even: `tate (l|r|o)`

<br>

NB: I realize `display_deform` uses the value `normal` to turn it off. Of course this should be consistent. I like `off` better (because shorter) – perhaps `off` could be an alias for `normal`?

**Comment by MBilderbeek** [comment]
I would propose to add it as a new option to `display_deform`. Can some GL shader person have some fun with this? :)

**Comment by MBilderbeek** [comment]
On the other hand: it would be fun to be able to have both display_deform 3D and a 90 degree rotation :)

**Comment by Colpocorto** [comment]
I opened a duplicate ticket since I wasn't aware of this one (I searched for the topic but this didn't pop up).

So, what's the status of this? Is it going to be done?  In my opinion, this shouldn't be overcomplicated. Just an option (available from both the console and the UI menu) to set it to horizontal or vertical (just like MAME or other arcade emulators).  

Probably 'vertical CW' and 'vertical CCW' options would be enough. Other approaches like arbitrary rotation angle or 3D deform are, in my opinion, a bigger development effort and not so valuable after all.

(It would be great if this setting could be set as a per-game basis and do some kind of auto-detection or hash-based detection... but of course this feature would be low priority).

**Comment by MBilderbeek** [comment]
It's a bit of a strange question, "is it going to be done". As you noticed, there are hundreds of open issues. Which one will be done (first) is a personal choice. If you think this is very interesting or important, then I invite you to try it yourself, or find someone who wants to help. A Pull Request is always appreciated and we'll gladly help anyone trying to implement stuff into openMSX.

Technically speaking, there may be some complicated factors of assumptions on window size(s), like for the main menu bar to fit or not, scalers, and who knows what else. Variable window size has been spoken of for years, but there have not been taken many concrete steps in that direction.
These assumptions may cause problems when rotating the output and there with changing the window size.

**Comment by Colpocorto** [comment]
Fair enough. Sorry if the question looked a bit demanding, wasn't on purpose.
It is just that the ticket was first created on 2017 and I wasn't sure if the request had been dismissed or it was still something on a low priority backlog.

Thank you for the good work.

**Comment by MBilderbeek** [comment]
If an issue is open for a long time it doesn't mean it isn't a good idea and it also doesn't mean we'll never implement it. It just means that no one has decided to work on it (yet). 

Still, if you can ask someone (or yourself) to help contributing, it is highly appreciated.
